### DISCUSSION TITLE
Dynamic Flow Modifiers for Enhanced Control

### DISCUSSION THREAD

**Discussion by n-yudin** [OrcaSlicer/OrcaSlicer#4983] [Ideas]
Hello,

I've been experimenting with dynamic flow control and believe it would greatly enhance Orca Slicer's functionality to enable flow rate changes through part modifiers.

![IMG_2262](https://github.com/SoftFever/OrcaSlicer/assets/166829349/69a1abb0-4e95-48c2-b389-39c91a20743f)
![IMG_2260](https://github.com/SoftFever/OrcaSlicer/assets/166829349/ea8a5660-e82a-41d2-817e-2c94f99a109b)

Currently, applying a modifier to a part and adjusting the flow rate doesn't yield the expected results. However, through experimentation, we've discovered that modifying the flow rate after generating the G-code produces intriguing outcomes. This feature is particularly beneficial for TPU materials, allowing for the adjustment of structural density or the creation of gradual transitions from soft to firm zones.

While similar results can be achieved by adjusting the infill percentage, this often results in the division of the part into separate zones, which may not always be desirable. Likewise, altering the infill wall width can change the infill structure and inadvertently divide the part into distinct zones.

![IMG_2265](https://github.com/SoftFever/OrcaSlicer/assets/166829349/7b6d170e-5ec1-44ce-b0fd-821a9a984d8c)
![IMG_2259](https://github.com/SoftFever/OrcaSlicer/assets/166829349/94e04532-6c78-4409-a016-bd488ba16d08)

Flow rate modifiers hold significant potential for rigid parts as well, enabling the creation of tougher zones by increasing material deposition without altering the part's geometry. This approach allows for wider infill walls while maintaining the part's integrity.

I believe integrating dynamic flow modifiers into Orca Slicer would unlock new possibilities for fine-tuning print settings and optimizing part performance.

https://github.com/SoftFever/OrcaSlicer/assets/166829349/2e4edab9-c506-4963-82b6-cdca9094fc18

Best regards, Nikolay Yudin

**Comment by WilliamJDauncey** [comment]
This would be a huge development for the 3D printing world, something would Orca slicer could pioneer too.

I completely support Nikolay’s impressive work, it would not only provide users with a whole new level of control over their prints, but would democratise 3D printing meta materials like Nikolay’s work for both consumer and industrial applications.

Currently there is no easy, consumer friendly way of achieving this without spending a huge amount of capitol which only very few industry giants can afford.


Nikolay’s work would give power back to individuals, democratising what has been gate keepen from users for so long

**Comment by Juanolo** [comment]
I think it would be a breakthrough for the use of 3D printing in clinical settings.

I fully support Nikolay's work. Please implement this feature!

Thanks to Nikolay and his team for volunteering their efforts to improve the software.

**Comment by PedorthicArt** [comment]
Hello,

Nikolay's work could be the basis for creating gradient stiffness infills that have a lot of practical applications - in my case insoles / foot orthosis. This is why I fully support his work and I hope in a near future to apply it in my design application! I'm sure it will be a great step forward for an open-source slicer and I'm sure if Orca Slicer does not implement it there will be for sure others who will implement it because it is a potential evolution/step forward. So the problem now is not when it will be implemented but who will be the first to implement it! I'm using only Prusa Slicer but for this option available I'll try/learn other slicers that has implemented it!! Will Orca lose this opportunity or will take Nikolay's work advantage??

**Comment by pillopaolo** [comment]
FULLY support your request! And should be very easy to implement: modifier management is there, just a matter of adding "Flow%" as a parameter and apply this as multiplication factor to the E move.  Do not understand why no slicer is implementing this.

**Reply by n-yudin** [reply]
Totally agree!

**Comment by Val-dose-things** [comment]
OMG yes please. this would help with printing at higher speeds in the infill well keeping the same strength. have issues with that all the time.
